File: java/src/main/java/com/brasibturger/models/MenuBurger.java
package com.brasibturger.models;

import lombok.*;
import jakarta.persistence.*;

@Entity
@Table(name = "menu_burger")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MenuBurger {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idMenuBurger;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_menu", nullable = false)
    private Menu menu;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_burger", nullable = false)
    private Burger burger;
}

File: csharp/BrasilBurger.Web/Models/MenuBurger.cs
namespace BrasilBurger.Web.Models;

public class MenuBurger
{
    public int Id { get; set; }
    public int IdMenu { get; set; }
    public int IdBurger { get; set; }
    
    public Menu? Menu { get; set; }
    public Burger? Burger { get; set; }
}
